packages/lesswrong/server/callbacks/postCallbacks.ts

```diff
@@ -689,3 +693,22 @@ getCollectionHooks("Posts").editSync.add(async function removeFrontpageDate(
   }
   return modifier;
 });
+
+getCollectionHooks("Posts").updateAfter.add(async function createJargonTermsCallback (_post, {newDocument, oldDocument, context}: UpdateCallbackProperties<"Posts">) {
+  const currentUser = context.currentUser
+  if (!currentUser) return _post
+  if (currentUser._id !== newDocument.userId) return _post
+  if (!userCanCreateAndEditJargonTerms(currentUser)) return _post
+
+  const newContents = await Revisions.findOne({_id: newDocument.contents_latest})
+  if (!newContents?.html) {
+    return _post;
+  }
+  const existingJargon = await JargonTerms.find({postId: newDocument._id}).fetch()
+  const changeMetrics = newContents.changeMetrics
+
+  if (changeMetrics.added > 1000 || !existingJargon.length) {
+    void createNewJargonTerms(newDocument._id, currentUser)
```

Comment:
When the user edits a post and saves, do we want to return to the user before `createNewJargonTerms` necessarily finishes running?